darts/tests/test_global_forecasting_models.py

```diff
@@ -113,3 +113,160 @@ def test_covariates(self):
+                    ts_pass_train_1 = adjust_type_fn(self.ts_pass_train_1.values())
+                    
+                    dummy_input = module.stack([ts_pass_train, ts_pass_train_1], axis=0)
+                    
+                    # (samples, TSlen, 1)
+                    pred_list = model.predict_from_dataset(n=PRED_LEN, input_series_dataset=dummy_input)
+                    
+                    
+                    self.assertTrue(pred_list.shape[0] == 2, 'Model {} did not return a list of prediction'.format(model_cls))
+                    
+                    for pred in pred_list:
+                        # turn into a TS
+                        pred_ts = TimeSeries.from_times_and_values(self.ts_pass_val.time_index(), to_np_fn(pred))
+                        
+                        mape_err = mape(self.ts_pass_val, pred_ts)
+                        self.assertTrue(mape_err < err, 'Model {} produces errors too high (several time series 2). '
+                                                        'Error = {}'.format(model_cls, mape_err))
+                    
+    
+        def test_multivariates_and_covariates_numpy(self):
+            
+            for model_cls, kwargs, err in models_cls_kwargs_errs:
+                if model_cls == NBEATSModel:
+                    # N-BEATS does not support multivariate
+                    continue
+                    
+                PRED_LEN = len(self.ts_pass_val)
+                ts_pass_train_array = self.ts_pass_train.values()
+                ts_pass_train_array_1 = self.ts_pass_train_1.values()
+                
+                multivariate_array = np.concatenate([ts_pass_train_array, ts_pass_train_array_1], axis=1) # (108, 2)
+                covariate_array = self.time_covariates_train.values() # (108, 2)
+                
+                input_array = np.concatenate([multivariate_array, covariate_array], axis=1) # (108, 4)
+                input_array = np.expand_dims(input_array, axis=0) # (1, 108, 4) 1 (multivariate TS) with 2 covariates
+                
+                model = model_cls(input_chunk_length=IN_LEN, output_chunk_length=OUT_LEN, **kwargs)
+                model.fit(series=[self.ts_pass_train.stack(self.ts_pass_train_1)],
+                          covariates=[self.time_covariates_train])
+                
+                pred = model.predict_from_dataset(n=PRED_LEN, input_series_dataset=input_array)
+                
+                # checking the overall returned shape
+                expected_shape = (input_array.shape[0], PRED_LEN, multivariate_array.shape[1]) # (n_samples, prediction_length, prediction_width)
+                self.assertEqual(expected_shape, pred.shape, 'Model prediction size not as expected.'
+                                 'Expected shape: {}, current shape: {}'.format(expected_shape, pred.shape))
+             
+        def test_multivariates_and_covariates_torch(self):
```

Comment:
I guess you should be able to remove most of these tests if you implement the changes I suggest and drop tensors and array support :D